unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -7275,6 +7223,160 @@ ProceduralAsset asset
             actionFinishedEmit(true, geoList);
         }
 
+        private Bounds GetAgentBounds(GameObject gameObject, Type agentType) {
+            Debug.Log(agentType);
+            Debug.Log(typeof(StretchAgentController));
+            Bounds bounds = new Bounds(gameObject.transform.position, Vector3.zero);
+            MeshRenderer[] meshRenderers = gameObject.GetComponentsInChildren<MeshRenderer>();
+            if (agentType == typeof(LocobotFPSAgentController)) {
+                meshRenderers = this.baseAgentComponent.BotVisCap.GetComponentsInChildren<MeshRenderer>();
+            } else if (agentType == typeof(StretchAgentController)) {
+                meshRenderers = this.baseAgentComponent.StretchVisCap.GetComponentsInChildren<MeshRenderer>();
+            }
+            foreach (MeshRenderer meshRenderer in meshRenderers) {
+                bounds.Encapsulate(meshRenderer.bounds);
+            }
+            return bounds;
+        }
+
+
+
+        public void spawnAgentBoxCollider(GameObject agent, Type agentType, Vector3 scaleRatio, bool useAbsoluteSize = false, bool useVisibleColliderBase = false) {
+            // Store the current rotation
+            Vector3 originalPosition = this.transform.position;
+            Quaternion originalRotation = this.transform.rotation;
+
+            Debug.Log($"the original position of the agent is: {originalPosition:F8}");
+
+            // Move the agent to a safe place and align the agent's rotation with the world coordinate system
+            this.transform.position = new Vector3(originalPosition.x + 100f, originalPosition.y + 100f, originalPosition.z + 100f);
+            this.transform.rotation = Quaternion.identity;
+
+            Debug.Log($"agent position after moving it out of the way is: {this.transform.position:F8}");
+
+            // Get the agent's bounds
+            var bounds = GetAgentBounds(agent, agentType);
+
+            Debug.Log($"the global position of the agent bounds is: {bounds.center:F8}");
+
+            // Check if the spawned boxCollider is colliding with other objects
+            int layerMask = LayerMask.GetMask("SimObjVisible", "Procedural1", "Procedural2", "Procedural3", "Procedural0");
+            
+            Vector3 newBoxCenter = new Vector3(bounds.center.x - 100f, bounds.center.y - 100f, bounds.center.z - 100f);
+            Debug.Log($"the center for the new box should be at the agent's original position but is: {newBoxCenter:F8}");
```

Comment:
Not necessarily true `Bounds.Encapsulate` might find a more fitting box center after adding the mesh geo to encapsulate.